### DISCUSSION TITLE
Set env variables in tests

### DISCUSSION THREAD

**Discussion by ewan-chalmers** [thunderclient/thunder-client-support#509] [Ideas]
In Postman, I can set variables in tests. This facilitates chaining a set of requests together, e.g.
1. Authenticate. Extract token from response header or body
2. Invoke query with token. Extract foo from JSON
3. Invoke query with token and foo

Is there any plan to provide similar capability in thunder-client?

**Comment by rangav** [comment]
Hi @ewan-chalmers 

We already have this functionality to set variables from json response, see details below
https://github.com/rangav/thunder-client-support#setenv

**Reply by ewan-chalmers** [reply]
That's great thanks @rangav . Don't know how I missed it :/

**Reply by rangav** [reply]
No problem, let me know if you have any further questions.

**Comment by alexeenkoff** [comment]
Great job, really!
Love your plugin!
What about possibility to set variables from json response to separate file `.env` or `.json`? 
My case - saving of bearer token for next requests. Set it to an active env collection means that team will need to see change of bearer token on every pull/merge requests. I think, that putting to and reading from some file from `.gitignore` can solve this problem

**Comment by rangav** [comment]
Hi @alexeenkoff 

you can save the token to local environment so it's not shared with others.

use format `{{token, local}}` in tests

set env variable with scope 
https://github.com/rangav/thunder-client-support#set-env-with-scope

all environments available 
https://github.com/rangav/thunder-client-support#environments

**Reply by alexeenkoff** [reply]
Hi, @rangav  
 `format {{token, local}}` is great. 
Would you plan to split `thunderEnvironment.json` to `thunderLocalEnvironment.json` and `thunderGlobalEnvironment.json`in next releases? It will allow to add `thunderLocalEnvironment.json` to `.gitignore`

**Comment by rangav** [comment]
`thunderLocalEnvironment.json` is already in separate file and not part of `thunderEnvironment.json`, so you don’t have to worry about `gitignore`.


Hope it’s clear now.

**Comment by rangav** [comment]
The values with format `{{token, local}}` will not be saved in `thunderEnvironement.json`
